dg/onboarding-guide.md

```diff
@@ -0,0 +1,292 @@
+# Table of Contents
+
+1. [Introduction to Tools and Frameworks used in CATcher](#introduction-to-tools-and-frameworks-used-in-catcher)
+    1. [Introduction to Angular](#introduction-to-angular)
+    2. [Introduction to Electron](#introduction-to-electron)
+        1. [How Electron Works](#how-electron-works)
+    3. [Introduction to GraphQL](#introduction-to-graphql)
+    4. [Authentication in CATcher](#authentication-in-catcher)
+2. [Introduction to Tests in CATcher](#introduction-to-tests-in-catcher)
+    1. [Jasmine Style Guide](#jasmine-style-guide)
+    2. [Angular TestBed Utility](#angular-testbed-utility)
+    3. [E2E Testing](#e2e-testing)
+3. [Workflow of CATcher](#workflow-of-catcher)
+4. [Getting Started](#getting-started)
+    1. [Setting Up](#setting-up)
+    2. [Commands](#commands)
+    3. [Good First Issues](#good-first-issues)
+    4. [Releasing a New Version of CATcher](#releasing-a-new-version-of-catcher)
+5. [Future Developments](#future-developments)
+
+# Introduction to Tools and Frameworks used in CATcher
+
+CATcher is a cross-platform application that utilises several tools. It is recommended that you understand how each tool comes into play before you start working on CATcher.
+
+## Introduction to Angular
+
+Angular is a TypeScript-based web application framework. CATcher is built using Angular and hence our codebase is in Typescript. It is recommended that you familiarise yourself with Typescript as well as Angular before you attempt to work on CATcher.
+
+### Angular Structure
+
+An Angular application is largely made up of Components, Services, and the corresponding HTML and CSS files for each Component. A Component is a direct representation of visible aspects of an application, while a Service provides useful functionalities to Components where needed. The separation of components and services increases modularity and reusability, as through dependency injection (DI), the service class can provide services to different parts of the application.
+
+For more details on the structure of Angular, visit the Angular Guide via the links below.
+
+<question type="text" header="What is the decorator used to define a service class?"
+          keywords="injectable" threshold="0.5" answer="@Injectable()">
+  <div slot="hint">
+
+  Read up on the [Angular Guide - Services and Dependency Injection](https://angular.io/guide/architecture#services-and-dependency-injection) for more details on services and dependency injection.
```

Comment:
For Angular shall we get the new devs to walkthrough Angular's tutorial to get an understanding of it. https://angular.io/tutorial
So like do the tutorial then try the question?

It was what we did when I first joined the project to learn Angular 😆

Reply:
Yes sure, I can add that into the docs! The tutorials we give them will likely not be enough😭

Reply:
Yeah but thats okay though I think as long as we can point them to some possible resources to refer to. Can't expect this page to be a "module" on web development 😆

Reply:
Yeah, I agree! Just a few will be enough imo, just some checkpoints for students to check their understanding aft visiting the resources (I think that's what Prof. Damith intended for us to do!)